File: AppliedProgramming_TheBigRewrite/Exercise03_02.py
# Rewrite 03_02

reordered_file = []
with open('ex3_data/gene_positions1.txt', 'r') as gene_position1_in:
    for line in gene_position1_in:
        line_list = line.split()
        reordered_line = [line_list[2], line_list[0], line_list[1], line_list[3], line_list[4]]
        reordered_line = '\t'.join(reordered_line) + '\n'
        reordered_file.append(reordered_line)

with open('ex3_data/gene_position1_reordered.txt', 'w') as gene_position1_out:
    gene_position1_out.writelines(reordered_file)

File: AppliedProgramming03/Exercise03_02.pl
# Write a program that reads the file “gene_positions1.txt” and reorders the columns
# to be “chromosome start end name score”. Print the result to screen.

use strict;
use warnings;

# Absolute file path, fix later
my $input_file =
'ex3data\gene_positions1.txt';

my @output_file;
open( my $fh_input, '<', $input_file )
  or die "Cannot read from file $input_file. Error: $!.\n";

while ( my $file_line = <$fh_input> ) {
	chomp $file_line;
	my @input_line = split( /\t/, $file_line )
	  ;    # Splits the current line at tab and saves the result in an array
	push( @output_file,
		    $input_line[2] . "\t"
		  . $input_line[0] . "\t"
		  . $input_line[1] . "\t"
		  . $input_line[3] . "\t"
		  . $input_line[4]
		  . "\n" );    # Rearranges the different columns in a new array
}
close($fh_input);
print
"The following is the edited content of the file $input_file that will overwrite its original content:\n\n@output_file\n";

open( my $fh_output, '>', $input_file )
  or die "Cannot write to file $input_file. Error: $!.\n";

print $fh_output "@output_file";
close($fh_output);

print "Done\nThank you for using this application!\n";